source/WB-CAN-Firmware.c

```diff
@@ -233,38 +354,58 @@ static void ControlTask(void *pv) {
 
 	// Set Pin to ON State
 	//GPIO_PortSet(valvePin, valvePinMask);
-	GPIO_PortClear(valvePin, valvePinMask);
+	GPIO_PortClear(VALVE_PIN, VALVE_PIN_MASK);
 	// PID Loop:
 	// Mode: P = 1, PI = 2
 	uint32_t mode = 1;
-	// Reading from Potentiometer:
-	float desired_val = 1.700; // In ATM
-	// Reading from Sensor:
-	//float desired_val = 20;
 
-	float error;
-	float sensor = adcRead(adc16ConfigStruct, adc16ChannelConfigStruct);
+
+	static char data_out[80];
+	float error, desired_val, kp;
+
+	float sensor = adcRead(adc16ConfigStruct, adc16ChannelConfigStruct); // Reads in Voltage
+
 	float out;
 	// Integral Components:
 	float sensor_old;
 	float integral;
 	float sample_time = 0.5;
 
 	float ki = 0.1;
-	float kp = 150.0;
+
 	int duty_offset = 50;
 
+	if(testConfig == KULITE){
+		// Reading from the Sensor
+		desired_val = 10.0; // In ATM
+		kp = 150.0;
+		sensor = sensor * pressureScaling; // To convert Volts to ATM
+	}
+	else if (testConfig == POTENTIOMETER){
+		// Reading from Potentiometer:
+		desired_val = 1.5;
+		kp = 1;//50;
+	}
+	else {
+		PRINTF("NO CONFIG SETUP");
+		for(;;);
+	}
+
 	while (1)
 	{
 		vTaskDelay(pdMS_TO_TICKS(sample_time*1000)); // Delay in milliseconds
 		sensor_old = sensor;
 		sensor = adcRead(adc16ConfigStruct, adc16ChannelConfigStruct);
 
+
+		sprintf(data_out, "%f\t\t %d\r", sensor*pressureScaling, duty_cycle);
+		SEGGER_RTT_WriteString(0, data_out);
+
 		error = desired_val - (sensor);
 		out = (int) (kp * error);
 
 
-		if (mode == 2){ // PI Controller:
+		if (mode == 2){ // PI Controller: Have not Tested
```

Comment:
consider mode as an enum too, and maybe a switch statement... sensing a pattern?

Reply:
Done, looks a lot cleaner. I left the `if` statement within the switch so that it wouldn't look like so many cases.